web3deployment/deployerv3.js

```diff
@@ -0,0 +1,473 @@
+const fs = require("fs");
+const path = require('path');
+const RLP = require('rlp');
+const BigNumber = require('bignumber.js')
+
+process.on('unhandledRejection', console.error.bind(console))
+
+const { configPath, gasPriceGwei, printPrivateKey, rpcUrl, signedTxOutput, dontSendTx, chainId: chainIdInput } = require('yargs')
+    .usage('Usage: $0 --config-path [path] --gas-price-gwei [gwei] --print-private-key [bool] --rpc-url [url] --signed-tx-output [path] --dont-send-tx [bool] --chain-id')
+    .demandOption(['configPath', 'gasPriceGwei', 'rpcUrl'])
+    .boolean('printPrivateKey')
+    .boolean('dontSendTx')
+    .argv;
+const web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
+const solc = require('solc')
+
+const rand = web3.utils.randomHex(7);
+const privateKey = web3.utils.sha3("js sucks" + rand);
+if (printPrivateKey) {
+  console.log("privateKey", privateKey);
+  let path = "privatekey_"  + web3.utils.randomHex(7) + ".txt";
+  fs.writeFileSync(path, privateKey, function(err) {
+      if(err) {
+          return console.log(err);
+      }
+  });
+}
+const account = web3.eth.accounts.privateKeyToAccount(privateKey);
+const sender = account.address;
+const gasPrice = BigNumber(gasPriceGwei).mul(10 ** 9);
+const signedTxs = [];
+let nonce;
+let chainId = chainIdInput;
+
+console.log("from",sender);
+
+async function sendTx(txObject) {
+  const txTo = txObject._parent.options.address;
+
+  let gasLimit;
+  try {
+    gasLimit = await txObject.estimateGas();
+  }
+  catch (e) {
+    gasLimit = 500 * 1000;
+  }
+
+  if(txTo !== null) {
+    gasLimit = 500 * 1000;
+  }
+
+  //console.log(gasLimit);
+  const txData = txObject.encodeABI();
+  const txFrom = account.address;
+  const txKey = account.privateKey;
+
+  const tx = {
+    from : txFrom,
+    to : txTo,
+    nonce : nonce,
+    data : txData,
+    gas : gasLimit,
+    chainId,
+    gasPrice
+  };
+
+  const signedTx = await web3.eth.accounts.signTransaction(tx, txKey);
+  nonce++;
+  // don't wait for confirmation
+  signedTxs.push(signedTx.rawTransaction)
+  if (!dontSendTx) {
+    web3.eth.sendSignedTransaction(signedTx.rawTransaction, {from:sender});
+  }
+}
+
+async function deployContract(solcOutput, contractName, ctorArgs) {
+
+  const actualName = contractName;
+  const bytecode = solcOutput.contracts[actualName].bytecode;
+
+  const abi = solcOutput.contracts[actualName].interface;
+  const myContract = new web3.eth.Contract(JSON.parse(abi));
+  const deploy = myContract.deploy({data:"0x" + bytecode, arguments: ctorArgs});
+  let address = "0x" + web3.utils.sha3(RLP.encode([sender,nonce])).slice(12).substring(14);
+  address = web3.utils.toChecksumAddress(address);
+
+  await sendTx(deploy);
+
+  myContract.options.address = address;
+
+  return [address,myContract];
+}
+
+const contractPath = path.join(__dirname, "../contracts/");
+
+const input = {
+  "ConversionRatesInterface.sol" : fs.readFileSync(contractPath + 'ConversionRatesInterface.sol', 'utf8'),
```

Comment:
please create a separate file that exports this input data.

I found myself fixing it so many times.

So it can be reused many times and fixed in one central place.

or if we have any other idea to use this list in all deploy scripts with min work

Reply:
Ahhhh.... that's true. Ok, good idea to put it in a separate script.

Reply:
I'm not sure how that would work if we have both v4 and v5 versions though... Hmm

Reply:
well
at least have the v4 inputs in a separate input file.
We will have to worry about v5 later.